python/ribasim/ribasim/model.py

```diff
@@ -264,10 +264,10 @@ def write(self, filepath: str | PathLike[str]) -> Path:
         # TODO
         # self.validate_model()
         filepath = Path(filepath)
+        self.filepath = filepath
```

Comment:
We could make `filepath` optional for `write`, though then we are encouraging overwriting models, and it would perhaps not be obvious to code readers where their model ended up.

This feels ok to me since once an in-memory model is written to disk, it has a `filepath`. Otherwise that would only be set when reading a model with `Model.read("path/to/ribasim.toml")`.